src/providers/list/superstream/index.ts

```
@@ -22,252 +22,255 @@ const nanoid = customAlphabet("0123456789abcdef", 32);
+      seasonCount: detailRes?.season?.length,
+    } as MWProviderMediaResult;
+  },
 
-	async searchForMedia(query: MWQuery): Promise<MWProviderMediaResult[]> {
-		const apiQuery = {
-			module: "Search3",
-			page: "1",
-			type: "all",
-			keyword: query.searchQuery,
-			pagelimit: "20",
-		};
-		const searchRes = (await get(apiQuery, true).then((r) => r.json())).data;
+  async searchForMedia(query: MWQuery): Promise<MWProviderMediaResult[]> {
+    const apiQuery = {
+      module: "Search3",
+      page: "1",
+      type: "all",
+      keyword: query.searchQuery,
+      pagelimit: "20",
+    };
+    const searchRes = (await get(apiQuery, true).then((r) => r.json())).data;
 
-		const movieResults: MWProviderMediaResult[] = (searchRes || [])
-			.filter((item: any) => item.box_type === 1)
-			.map((item: any) => ({
-				title: item.title,
-				year: item.year,
-				mediaId: item.id,
-			}));
-		const seriesResults: MWProviderMediaResult[] = (searchRes || [])
-			.filter((item: any) => item.box_type === 2)
-			.map((item: any) => ({
-				title: item.title,
-				year: item.year,
-				mediaId: item.id,
-				seasonId: "1",
-				episodeId: "1",
-			}));
+    const movieResults: MWProviderMediaResult[] = (searchRes || [])
+      .filter((item: any) => item.box_type === 1)
+      .map((item: any) => ({
+        title: item.title,
+        year: item.year,
+        mediaId: item.id,
+      }));
+    const seriesResults: MWProviderMediaResult[] = (searchRes || [])
+      .filter((item: any) => item.box_type === 2)
+      .map((item: any) => ({
+        title: item.title,
+        year: item.year,
+        mediaId: item.id,
+        seasonId: "1",
+        episodeId: "1",
+      }));
 
-		if (query.type === "movie") {
-			return movieResults;
-		}
-		return seriesResults;
-	},
+    if (query.type === MWMediaType.MOVIE) {
+      return movieResults;
+    }
+    if (query.type === MWMediaType.SERIES) {
+      return seriesResults;
+    }
+    return [];
```

I'd throw an error here, something like "invalid media type inputted" if it doesnt match movie nor series

---

Ok